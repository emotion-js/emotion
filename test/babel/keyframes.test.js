import * as babel from 'babel-core'
import plugin from '../../src/babel'
import * as fs from 'fs'
jest.mock('fs')

fs.existsSync.mockReturnValue(true)

describe('babel keyframes', () => {
  describe('inline', () => {
    test('keyframes basic', () => {
      const basic = `
        const rotate360 = keyframes\`
          from {
            transform: rotate(0deg);
          }
          to {
            transform: rotate(360deg);
          }
      \`;`
      const { code } = babel.transform(basic, {
        plugins: [[plugin]]
      })
      expect(code).toMatchSnapshot()
    })
    test('keyframes with interpolation', () => {
      const basic = `
        const rotate360 = keyframes\`
          from {
            transform: rotate(0deg);
          }
          to {
            transform: rotate(\${endingRotation});
          }
      \`;`
      const { code } = babel.transform(basic, {
        plugins: [[plugin]]
      })
      expect(code).toMatchSnapshot()
    })
  })
  describe('extract', () => {
    test('keyframes basic', () => {
      const basic = `
        const rotate360 = keyframes\`
          from {
            transform: rotate(0deg);
          }
        
          to {
            transform: rotate(360deg);
          }
      \`;`
      const { code } = babel.transform(basic, {
        plugins: [[plugin, { extractStatic: true }]],
        babelrc: false,
        filename: __filename
      })
      expect(code).toMatchSnapshot()
      expect(fs.writeFileSync).toHaveBeenCalledTimes(1)
      expect(fs.writeFileSync.mock.calls[0][1]).toMatchSnapshot()
    })
    test('keyframes with interpolation', () => {
      const basic = `
        const rotate360 = keyframes\`
          from {
            transform: rotate(0deg);
          }
        
          to {
            transform: rotate(\${endingRotation});
          }
      \`;`
      const { code } = babel.transform(basic, {
        plugins: [[plugin, { extractStatic: true }]],
        babelrc: false,
        filename: __filename
      })
      expect(code).toMatchSnapshot()
      expect(fs.writeFileSync).toHaveBeenCalledTimes(1)
    })
  })
})
