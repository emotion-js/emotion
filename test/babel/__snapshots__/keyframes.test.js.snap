// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel keyframes extract keyframes basic 1`] = `
"import \\"./keyframes.test.emotion.css\\";

const rotate360 = \\"css-rotate360-f35ahc\\";"
`;

exports[`babel keyframes extract keyframes basic 2`] = `
"@-webkit-keyframes css-rotate360-f35ahc {
    from {
        -webkit-transform: rotate(0deg);
        transform: rotate(0deg)
    }
    to {
        -webkit-transform: rotate(360deg);
        transform: rotate(360deg)
    }
}
@keyframes css-rotate360-f35ahc {
    from {
        -webkit-transform: rotate(0deg);
        transform: rotate(0deg)
    }
    to {
        -webkit-transform: rotate(360deg);
        transform: rotate(360deg)
    }
}"
`;

exports[`babel keyframes extract keyframes with interpolation 1`] = `
"
const rotate360 = /*#__PURE__*/keyframes([], [endingRotation], function createEmotionStyledRules(x0) {
  return [{
    \\"from\\": {
      \\"WebkitTransform\\": \\"rotate(0deg)\\",
      \\"transform\\": \\"rotate(0deg)\\"
    },
    \\"to\\": {
      \\"WebkitTransform\\": \`rotate(\${x0})\`,
      \\"transform\\": \`rotate(\${x0})\`
    }
  }];
});"
`;

exports[`babel keyframes inline keyframes basic 1`] = `
"
const rotate360 = /*#__PURE__*/keyframes([], [], function createEmotionStyledRules() {
  return [{
    \\"from\\": {
      \\"WebkitTransform\\": \\"rotate(0deg)\\",
      \\"transform\\": \\"rotate(0deg)\\"
    },
    \\"to\\": {
      \\"WebkitTransform\\": \\"rotate(360deg)\\",
      \\"transform\\": \\"rotate(360deg)\\"
    }
  }];
});"
`;

exports[`babel keyframes inline keyframes with interpolation 1`] = `
"
const rotate360 = /*#__PURE__*/keyframes([], [endingRotation], function createEmotionStyledRules(x0) {
  return [{
    \\"from\\": {
      \\"WebkitTransform\\": \\"rotate(0deg)\\",
      \\"transform\\": \\"rotate(0deg)\\"
    },
    \\"to\\": {
      \\"WebkitTransform\\": \`rotate(\${x0})\`,
      \\"transform\\": \`rotate(\${x0})\`
    }
  }];
});"
`;
